includes/push-syndicate-encryption.php

```diff
@@ -1,18 +1,72 @@
 <?php
 
+function push_syndicate_get_cipher() {
+	$cipher = 'aes-256-cbc';
+
+	if ( version_compare( PHP_VERSION, '7.1', '<' ) ) {
+		return MCRYPT_RIJNDAEL_256;
+	}
+
+	if ( in_array( $cipher, openssl_get_cipher_methods(), true ) ) {
+		return array(
+			'cipher' => $cipher,
+			'iv'     => substr( md5( md5( PUSH_SYNDICATE_KEY ) ), 0, 16 ),
+			'key'    => md5( PUSH_SYNDICATE_KEY ),
+		);
+	}
+
+	return false; // @TODO: return another default cipher? return exception?
+}
+
 function push_syndicate_encrypt( $data ) {
+	// @todo: replace mcrypt with openssl. problem: Rijndael AES is not available on openssl;s AES-256.
+	// Will most likely break backwards compatibility with older keys
+	// https://stackoverflow.com/questions/49997338/mcrypt-rijndael-256-to-openssl-aes-256-ecb-conversion
+
+	// Backwards compatibility with PHP < 7.1. Use mcrypt instead.
+	if ( version_compare( PHP_VERSION, '7.1', '<' ) ) {
+		// @codingStandardsIgnoreStart
+		$data = serialize( $data );
+		return base64_encode(mcrypt_encrypt(MCRYPT_RIJNDAEL_256, md5(PUSH_SYNDICATE_KEY), $data, MCRYPT_MODE_CBC, md5(md5(PUSH_SYNDICATE_KEY))));
+		// @codingStandardsIgnoreEnd
```

Comment:
Why ignore all coding standards, when you could just ignore the ones we can't address?

```suggestion
		// phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize -- kept for BC.
		$data = serialize( $data );
		// phpcs:ignore PHPCompatibility.FunctionUse.RemovedFunctions.mcrypt_encryptDeprecatedRemoved,PHPCompatibility.Extensions.RemovedExtensions.mcryptDeprecatedRemoved,PHPCompatibility.Constants.RemovedConstants.mcrypt_rijndael_256DeprecatedRemoved,PHPCompatibility.Constants.RemovedConstants.mcrypt_mode_cbcDeprecatedRemoved
		return base64_encode( mcrypt_encrypt( MCRYPT_RIJNDAEL_256, md5( PUSH_SYNDICATE_KEY ), $data, MCRYPT_MODE_CBC, md5( md5( PUSH_SYNDICATE_KEY ) ) ) );
```

Reply:
Ah, I was just trying to avoid a huge line with the PHPCS ignores. But I changed it to be more specific, as suggested.